core/src/contracts/stablex_auction_element.rs

```diff
@@ -44,62 +40,29 @@ impl StableXAuctionElement {
         let denominator = BigEndian::read_u128(&bytes[80..96]);
         let remaining = BigEndian::read_u128(&bytes[96..112]);
         let id = u16::from_le_bytes([bytes[113], bytes[112]]);
-        let (buy_amount, sell_amount) = compute_buy_sell_amounts(numerator, denominator, remaining);
         StableXAuctionElement {
-            valid_from,
-            valid_until,
             sell_token_balance,
             order: Order {
                 id,
                 account_id,
                 buy_token,
                 sell_token,
-                buy_amount,
-                sell_amount,
+                numerator,
+                denominator,
+                remaining,
+                valid_from,
+                valid_until,
             },
         }
     }
 }
 
-pub fn compute_buy_sell_amounts(
-    numerator: u128,
-    denominator: u128,
-    remaining: u128,
-) -> (u128, u128) {
-    assert!(
-        remaining <= denominator,
-        "Smart contract should never allow this inequality"
-    );
-    // 0 on sellAmount (remaining <= denominator) is nonsense, but solver can handle it.
-    // 0 on buyAmount (numerator) is a Market Sell Order.
-    let buy_amount = if denominator > 0 {
-        // up-casting to handle overflow
-        let top = U256::from(remaining) * U256::from(numerator);
-        (top).ceiled_div(U256::from(denominator)).as_u128()
-    } else {
-        0
-    };
-    (buy_amount, remaining)
-}
-
 #[cfg(test)]
 pub mod tests {
     use super::*;
 
     fn emptyish_auction_element() -> StableXAuctionElement {
```

Comment:
nit: I would argue we could just remove `emptyish_auction_element` in favour of `Default::default()`.